libraries/shared/executor/executor_test.go

```diff
@@ -0,0 +1,72 @@
+package executor_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/executor"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/mocks"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/watcher"
+	"github.com/vulcanize/vulcanizedb/pkg/core"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+	"github.com/vulcanize/vulcanizedb/pkg/fakes"
+	"github.com/vulcanize/vulcanizedb/test_config"
+	"time"
+)
+
+var _ = Describe("Executor", func() {
+	var (
+		db         *postgres.DB
+		blockChain *fakes.MockBlockChain
+		mockPlugin fakes.MockPlugin
+	)
+	BeforeEach(func() {
+		db = test_config.NewTestDB(core.Node{ID: "testNode"})
+		blockChain = fakes.NewMockBlockChain()
+		mockPlugin = fakes.NewMockPlugin()
+	})
+
+	It("adds transformer sets to the executor", func() {
+		ew := watcher.NewEventWatcher(db, blockChain, false, time.Second)
+		sw := watcher.NewStorageWatcher(mocks.NewMockStorageFetcher(), db, time.Second, time.Second)
+		cw := watcher.NewContractWatcher(db, blockChain, time.Second)
+		ex := executor.NewExecutor(&mockPlugin, &ew, sw, &cw)
+
+		loadErr := ex.LoadTransformerSets()
+		Expect(loadErr).NotTo(HaveOccurred())
+
+		Expect(ex.EthEventInitializers).To(HaveLen(len(mockPlugin.FakeEventInitializers)))
+		Expect(ex.EthStorageInitializers).To(HaveLen(len(mockPlugin.FakeStorageInitializers)))
+		Expect(ex.EthContractInitializers).To(HaveLen(len(mockPlugin.FakeContractInitializers)))
+		// the failure shows that expected equals received so not sure why the following doesn't work
```

Comment:
🤔 have you tried `ConsistOf`? Could be an object equality thing I suppose - or maybe a pointer/no-pointer mismatch?

Reply:
`ConsistOf` didn't work either. For example, `Expect(ex.EthEventInitializers).To(ConsistOf(mockPlugin.FakeEventInitializers[0]))` failed with the error

```
  Expected
      <[]transformer.EventTransformerInitializer | len:1, cap:1>: [0x45d1a80]
  to consist of
      <[]interface {} | len:1, cap:1>: [0x45d1a80]
```

The failure I get for the commented out code looks like
```
      <[]transformer.EventTransformerInitializer | len:1, cap:1>: [0x45cdf50]
  to equal
      <[]transformer.EventTransformerInitializer | len:1, cap:1>: [0x45cdf50]
```

🤷‍♀

Reply:
This was breaking my brain, so I went in a bit of a rabbit hole. I think this is happening because `TransformerInitializer`s are functions and it looks like `Gomega.Equal` can't be used to compare functions. Gomega uses go's `DeepEqual` to determine equality. And after digging into that [source code](https://github.com/golang/go/blob/master/src/reflect/deepequal.go#L130), I found this:

```go
switch v1.Kind() {
...
case Func:
	if v1.IsNil() && v2.IsNil() {
		return true
	}
	// Can't do better than this:
	return false
...
}
```

Reply:
Ohhh ok! Wow, thanks for figuring that out. I guess we'll have to settle for the length comparison then